genie-web/src/main/java/com/netflix/genie/web/health/MemoryHealthIndicator.java

```diff
@@ -0,0 +1,64 @@
+package com.netflix.genie.web.health;
+
+import com.sun.management.OperatingSystemMXBean;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+
+/**
+ * Health indicator for system memory usage. Also mark the service out-of-service if it crosses the threshold.
+ * @author amajumdar
+ * @since 3.0.0
+ */
+@Component
+@Slf4j
+public class MemoryHealthIndicator implements HealthIndicator {
+    private static final String FREE_PHYSICAL_MEMORY_SIZE = "freePhysicalMemorySize";
+    private static final String TOTAL_PHYSICAL_MEMORY_SIZE = "totalPhysicalMemorySize";
+
+    private final double maxUsedPhysicalMemoryPercentage;
+    private final OperatingSystemMXBean operatingSystemMXBean;
+
+    /**
+     * Constructor.
+     *
+     * @param maxUsedPhysicalMemoryPercentage The maximum physical memory threshold
+     * @param operatingSystemMXBean MX bean for operating system
+     */
+    @Autowired
+    public MemoryHealthIndicator(
+            @Value("${genie.threshold.used-physical-memory-percent:90}")
```

Comment:
We haven't used dashes in any of our property names. I would switch this to camel case for consistency. We might also want to group them all under a ```genie.health``` level like this could be ```genie.health.memory.usedPhysicalMemoryPercent```